crates/joining_code/src/validation.rs

```diff
@@ -47,11 +53,14 @@ pub fn validate_joining_code(
                 return Ok(ValidateCallbackResult::Valid);
             };
             // TODO: find a way to TryFrom a ref, to avoid cloning.
-            let mem_proof = Element::try_from((*mem_proof).clone())?;
+            let mem_proof = match Record::try_from((*mem_proof).clone()) {
+                Ok(r) => r,
+                Err(e) => return Err(wasm_error!(WasmErrorInner::Guest(e.to_string()))),
+            };
```

Comment:
Huh, it seems like this error mapping shouldn't be necessary. Is it? Why? If it is necessary then maybe we need to fix something in Holochain so it isn't. Is it so that the line number comes through in the error?

Reply:
yes it's so the line number comes through because backtraces aren't stable in rust @maackle

Reply:
It's bc there is no attribute to convert into the new wasm error, so without the error mapping I end up receiving the following error:
```
Go to Result | SerializedBytes | SerializedBytesError

`?` couldn't convert the error to `hdk::prelude::WasmError`
the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
the following implementations were found:
  <hdk::prelude::WasmError as From<Infallible>>
  <hdk::prelude::WasmError as From<error::HhaError>>
required because of the requirements on the impl of `FromResidual<Result<Infallible, hdk::prelude::SerializedBytesError>>` for `Result<(), hdk::prelude::WasmError>`rustcE0277
```

Reply:
That is weird, its using an `HhaError` type? That should not exist in this repo